alacritty_terminal/src/term/mod.rs

```diff
@@ -844,12 +847,6 @@ pub struct Term<T> {
 
     pub visual_bell: VisualBell,
 
-    /// Saved cursor from main grid.
-    cursor_save: Cursor,
-
-    /// Saved cursor from alt grid.
-    cursor_save_alt: Cursor,
```

Comment:
As mentioned previously, these are necessary. I'd recommend looking into XTerm to see if they're persisted across alt grid switches, but we definitely need at least one of these.

Reply:
Can you explain what their function is, looked to me like they were used to store and restore data for `cursor` but doing `mem::swap` now seems to have the same effect to me.

Reply:
If there’s a test we can do to show the difference then of course that would be welcome.

Reply:
If you send the terminal escape `ESC 7`, move the cursor and then send the escape `ESC 8`, the cursor should return to the position where it was when the `ESC 7` escape was sent. To do that we obviously need to store information about the cursor and that's what this is used for.

This can be used to save and restore the cursor position, charset and cell template.

Based on my testing it is not necessary (or intended?) to preserve this across alt grid boundaries, however it's still necessary to store it.

Reply:
Makes sense, I’ll see if I can make a ref-test and make sure it works consistently